openedx_external_enrollments/tests/api/v0/test_views.py

```diff
@@ -0,0 +1,689 @@
+            'Authorization': 'test-token-type test-access-token',
+        }
+
+        self.assertEqual(self.base._get_enrollment_headers(), expected_headers)
+        post_mock.assert_called_with(settings.EDX_ENTERPRISE_API_TOKEN_URL, data)
+
+        post_mock.return_value.ok = False
+        self.assertIsNone(self.base._get_enrollment_headers())
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            post_mock.side_effect = Exception('test-exception')
+            self.assertIsNone(self.base._get_enrollment_headers())
+            log_capture.check(
+                ('openedx_external_enrollments.api.v0.views', 'ERROR', 'Failed to get token: test-exception'),
+            )
+
+    def test_get_enrollment_url(self):
+        """Testing _get_enrollment_url method."""
+        expected_url = '{}/enterprise-customer/{}/course-enrollments'.format(
+            settings.EDX_ENTERPRISE_API_BASE_URL,
+            settings.EDX_ENTERPRISE_API_CUSTOMER_UUID,
+        )
+        self.assertEqual(
+            expected_url,
+            self.base._get_enrollment_url(course_settings={}),
+        )
+
+    def test_str(self):
+        """
+        EdxEnterpriseExternalEnrollment overrides the __str__ method,
+        this test that the method __str__ returns the right value.
+        """
+        self.assertEqual(
+            'edX',
+            self.base.__str__(),
+        )
+
+
+class EdxInstanceExternalEnrollmentTest(TestCase):
+    """Test class for EdxInstanceExternalEnrollment class."""
+
+    def setUp(self):
+        """Set test database."""
+        self.base = EdxInstanceExternalEnrollment()
+
+    @patch('openedx_external_enrollments.api.v0.views.get_user')
+    def test_get_enrollment_data(self, get_user_mock):
+        """Testing _get_enrollment_data method."""
+        data = {
+            'course_mode': 'first_mode',
+            'user_email': 'test_email',
+        }
+        course_settings = {
+            'external_course_run_id': 'test_course_run_id',
+            'external_enrollment_mode_override': 'test_mode',
+        }
+        expected_data = {
+            'user': 'test-username',
+            'mode': 'test_mode',
+            'is_active': True,
+            "course_details": {
+                "course_id": 'test_course_run_id'
+            }
+        }
+
+        user = Mock()
+        user.username = 'test-username'
+        get_user_mock.return_value = (user, '')
+
+        self.assertEqual(
+            self.base._get_enrollment_data(data, course_settings),
+            expected_data,
+        )
+
+        del course_settings['external_enrollment_mode_override']
+        expected_data['mode'] = data['course_mode']
+
+        self.assertEqual(
+            self.base._get_enrollment_data(data, course_settings),
+            expected_data,
+        )
+
+    def test_get_enrollment_headers(self):
+        """Testing _get_enrollment_headers method."""
+        expected_headers = {
```

Comment:
as method that fails and no more, as integration that logs the exception and create the enrollmentlog with the exception

Reply:
Should we have to test the case you mention? @andrey-canon

Reply:
Yes, but the fact is that classes were not included and will not be included in this pr. since  probably they will be removed in the next step of this work.